src/core_plugins/kibana/public/dashboard/dashboard.js

```diff
@@ -230,16 +237,6 @@ app.directive('dashboardApp', function ($injector) {
 
       $scope.$watch('model.query', $scope.updateQueryAndFetch);
 
-      $scope.$watchCollection(() => dashboardState.getAppState().$newFilters, function (filters = []) {
```

Comment:
Moving the kuery update logic into filter_bar_click_handler allowed me to remove these crappy watches from all our apps 😃